example.yml

```diff
@@ -0,0 +1,47 @@
+## Example filter rule for gatekeeper
+#
+# default deny
+---
+- Deny:
+    address: Any
+    port: Any
+    protocol: Any
+# allow local ipv4 network 192.168.0.1/16
+- Allow:
+    address:
+      Specif:
+        IpAddr:
+          addr: 192.168.0.1
+          mask: 16
+    port: Any
+    protocol: Any
+# allow some google domains using port 443
+- Allow:
+    # {mail.,}google.{com,co.jp}
+    address:
+      Specif:
+        Domain:
+          # regexp pattern
+          pattern: "(mail\\.)?google\\.((com)|(co\\.jp))"
```

Comment:
- maybe we should add `\A` at the beginning and `\z` at the end?
- maybe we should avoid capturing it? (but it may be hard to read).
- The single-quote makes it easier to not think about escaping? (but maybe it's confusing...)

<details>
    <summary><Pre>pattern: '\A(?:mail\.)?google\.(?:(?:com)|(?:co\.jp))\z'</pre></summary>

```yaml
- Deny:
    address: Any
    port: Any
    protocol: Any
- Allow:
    address:
      Specif:
        Domain:
          pattern: '\A(?:mail\.)?google\.(?:(?:com)|(?:co\.jp))\z'
    port:
      Specif: 443
    protocol:
      Specif: Tcp
```
</details>


https://rubular.com/r/ubSKHVNp2oUIUb

Reply:
I don't think it's that important to avoid capturing. This is just one example, so I hope it's easy to read.